File: python/claude/p13_nr_ver3.py
id_sequence = [10, 20, 30, 40, 50]

def get_value(index):
    if not isinstance(index, int) or isinstance(index, bool):
        return -1
    if index < 0 or index >= len(id_sequence):
        return -1
    return id_sequence[index]

def parse_index(raw):
    stripped = raw.strip()
    if not stripped:
        return None
    # Reject floats like "2.0" or "2e1"
    if not stripped.lstrip('-+').isdigit():
        return None
    try:
        return int(stripped)
    except ValueError:
        return None

def main():
    try:
        raw = input("Enter an index: ")
    except EOFError:
        print("Input error. Result: -1")
        return

    index = parse_index(raw)
    if index is None:
        print("Invalid input. Result: -1")
        return

    print(f"Result: {get_value(index)}")

main()

File: javascript/claude/p13_nr_ver3.js
const id_sequence = [10, 20, 30, 40, 50];

function getValue(index) {
    if (!Number.isInteger(index) || index < 0 || index >= id_sequence.length) return -1;
    return id_sequence[index];
}

function parseIndex(input) {
    const trimmed = input.trim();
    // Reject empty, floats, hex, scientific notation, whitespace-embedded
    if (!/^-?\d+$/.test(trimmed)) return null;
    const value = Number(trimmed);
    if (!Number.isInteger(value)) return null;
    return value;
}

const readline = require("readline");
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

rl.question("Enter an index: ", (input) => {
    const index = parseIndex(input);
    if (index === null) {
        console.log("Invalid input. Result: -1");
    } else {
        console.log("Result:", getValue(index));
    }
    rl.close();
});